### PR TITLE
Coercing only OptionSetValues to boolean for CountIf and LookUp

### PR DESCRIPTION
**Pull request by emhomm4**
Re-do of this PR: 
https://github.com/microsoft/Power-Fx/pull/187

But instead only coercing OptionSetValue.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/CountIf.cs [hunk 1]
```diff
@@ -64,10 +64,21 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
             var fValid = CheckInvocation(args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
             Contracts.Assert(returnType == DType.Number);
 
-            // Ensure that all the args starting at index 1 are booleans.
+            // Ensure that all the args starting at index 1 are booleans or coerece to boolean if they are OptionSetValues.
             for (var i = 1; i < args.Length; i++)
             {
-                if (!DType.Boolean.Accepts(argTypes[i]))
+                if (argTypes[i] == DType.OptionSetValue && CheckType(args[i], argTypes[i], DType.Boolean, DefaultErrorContainer, out var matchedWithCoercion))
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/CountIf.cs:70]: Why go through CheckType? We could just use `argTypes[i].CoercesTo(Dtype.Boolean)` at this point, I think.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Lookup.cs [hunk 2]
```diff
@@ -47,6 +48,23 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
             // The return type is dictated by the last argument (projection) if one exists. Otherwise it's based on first argument (source).
             returnType = args.Length == 2 ? argTypes[0].ToRecord() : argTypes[2];
 
+            // Ensure that the arg at index 1 is boolean or can coerece to boolean if they are OptionSetValues.
+            if (DType.Boolean.Accepts(argTypes[1]))
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Lookup.cs:52]: Empty code blocks are a little sketchy. Either you could `return fvalid` here, or just remove it, and change line 62 to `else if (!Dtype.Boolean.Accepts(ArgTypes[1])

**Review by lesaltzm — APPROVED** [review]

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/CountFunctions.txt [hunk 3]
```diff
@@ -23,4 +23,7 @@ Blank()
 #Error
 
 >> CountIf([1, 2, 3, 4], 123)
-Errors: Error 22-25: Invalid argument type (Number). Expecting a Boolean value instead.
+Errors: Error 22-25: Expected boolean. We expect a boolean (true/false) at this point in the formula.|Error 0-26: The function 'CountIf' has some invalid arguments.
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/CountFunctions.txt:26]: consider setting haspreciseerrors on the function, the 2nd error message isn't super useful

Comment (emhomm4) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/CountFunctions.txt]: Added

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/CountFunctions.txt [hunk 4]
```diff
@@ -23,4 +23,7 @@ Blank()
 #Error
 
 >> CountIf([1, 2, 3, 4], 123)
-Errors: Error 22-25: Invalid argument type (Number). Expecting a Boolean value instead.
+Errors: Error 22-25: Expected boolean. We expect a boolean (true/false) at this point in the formula.
+
+>> CountIf([1, 2, 3, 4], "abc")
+Errors: Error 22-27: Expected boolean. We expect a boolean (true/false) at this point in the formula.
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/CountFunctions.txt:29]: This is odd, where else is this error used? It makes me think it expects a boolean literal, but that's not the case. The InvalidArgumentType Error should be good enough here

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/CountFunctions.txt (hunk 3 above)
Comment (lesaltzm) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/CountFunctions.txt]: what about lookup?

**Review by lesaltzm — DISMISSED** [review]

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**